### DISCUSSION TITLE
Errors after upgrading to 4.0

### DISCUSSION THREAD

**Discussion by BasBBakker** [supabase/gotrue-csharp#59] [Q&A] [answered]
My code runs fine in 3.x.
When I upgrade gotrue- to 4.0 i get the error:

WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not resolve type with token 01000021 from typeref (expected class 'Supabase.Gotrue.ClientOptions`1' in assembly 'Gotrue, Version=3.1.1.0, Culture=neutral, PublicKeyToken=null')
System.TypeLoadException: Could not resolve type with token 01000021 from typeref (expected class 'Supabase.Gotrue.ClientOptions`1' in assembly 'Gotrue, Version=3.1.1.0, Culture=neutral, PublicKeyToken=null')
   at Program.<>c__DisplayClass0_0.<<Main>$>b__3(IServiceProvider provider) in C:\Users\bas\OneDrive - Dutch Duck Software\Github\SupabaseTester\SupabaseTester\SupabaseTester.Client\Program.cs:line 43
   at Microsoft.Extensions.DependencyI
(left out the rest of the error message)

The code it refers to is

builder.Services.AddScoped<Supabase.Client>(
    provider => new Supabase.Client(
        url,
        key,
        new Supabase.SupabaseOptions
        {
            AutoRefreshToken = true,
            AutoConnectRealtime = true,
            PersistSession = true,
            SessionHandler = new CustomSupabaseSessionHandler(
                provider.GetRequiredService<ILocalStorageService>(),
                provider.GetRequiredService<ILogger<CustomSupabaseSessionHandler>>()
            )
        }
    )
);

This is based on the 
https://github.com/supabase-community/supabase-csharp/tree/master/Examples/BlazorWebAssemblySupabaseTemplate
example.

Does anyone know how I can fix this error? 

Thanks a lot!

Bas

**Comment by BasBBakker** [comment]
The error refers to  in assembly 'Gotrue, Version=3.1.1..
But I have linked to version 4.0. I cannot figure why it gives the wrong link.
Thanks in advance for any help!

**Comment by acupofjose** [comment]
Unfortunately, `supabase-csharp` is behind on implementing the changes from `gotrue-csharp@4.0.0` - I'm waiting until supabase-community/realtime-csharp#28 and supabase-community/postgrest-csharp#65 are closed and merged as those will be major version changes as well.

**Comment by wiverson** [comment]
I ran into something unusual when working with a Unity build that might(?) be related. The call to assembly.GetCustomAttribute was failing, and so I wound up needing to replace it with this as a workaround.

```csharp
using System;
using System.Reflection;

namespace Supabase.Core {
    public static class Util {
        public static string GetAssemblyVersion(Type clientType) {
            var assembly = clientType.Assembly;
            string informationVersion = "Unknown";
            try {
                informationVersion = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
                    .InformationalVersion;
            } catch (Exception e) {
                Console.WriteLine(e);
            }

            var name = assembly.GetName().Name;

            return $"{name.ToString().ToLower()}-csharp/{informationVersion}";
        }
    }
}

```

**Comment by BasBBakker** [comment, marked as answer]
It works with version 0.10 of supabase c#. Thank you!
